ilastik/applets/batchProcessing/hbp.py

```diff
@@ -0,0 +1,137 @@
+"""HBP-specific utilities."""
+
+import functools
+import io
+import types
+from dataclasses import dataclass
+from typing import Mapping, Optional, Type, Union
+
+import h5py
+import numpy
+import requests
+
+from ilastik.applets.dataSelection.opDataSelection import OpDataSelectionGroup
+from ilastik.applets.featureSelection.opFeatureSelection import OpFeatureSelection
+from ilastik.config import cfg
+from ilastik.workflow import Workflow
+from ilastik.workflows import PixelClassificationWorkflow
+from ilastik.workflows.objectClassification.objectClassificationWorkflow import ObjectClassificationWorkflow
+
+
+CATEGORIES: Mapping[Type[Workflow], str] = types.MappingProxyType(
+    {PixelClassificationWorkflow: "pixel_classification", ObjectClassificationWorkflow: "object_classification"}
+)
+
+ORDERS: Mapping[str, int] = types.MappingProxyType(
+    {
+        "GaussianSmoothing": 0,
+        "LaplacianOfGaussian": 2,
+        "GaussianGradientMagnitude": 1,
+        "DifferenceOfGaussians": 0,
+        "StructureTensorEigenvalues": 1,
+        "HessianOfGaussianEigenvalues": 2,
+    }
+)
+
+
+def category(workflow: Workflow) -> str:
+    """Workflow category name, or empty string if HBP server does not support the workflow."""
+    return next((cat for cls, cat in CATEGORIES.items() if isinstance(workflow, cls)), "")
+
+
+def halo(*, scale: float, order: int) -> int:
+    """Compute halo size from scale and filter order."""
+    return int(3 * scale + 0.5 * order + 0.5)
```

Comment:
```suggestion
    return numpy.ceil((3 * order + 0.5 ) * scale)
```

personally I find the `int(... + 0.5)` trick for ceil too implicit, especially if the term already has `+ 0.5` somewhere.

see also [vigra](https://ukoethe.github.io/vigra/doc-release/vigra/classvigra_1_1Gaussian.html) and [fastfilters](https://github.com/svenpeter42/fastfilters/blob/715281e8ee20e15080e416b60e13e1d33984908f/src/library/fir_kernel.c#L77)